### PR TITLE
statistic: return user-visible scanned bytes

### PR DESCRIPTION
**Pull request by zhongzc**
### What problem does this PR solve?

Lack of some statistics for calculating SLI/SLO.

### What is changed and how it works?

What's Changed:

Collect processed data size in `PointGetter` and `Scanner`.

### Related changes

- kvproto: https://github.com/pingcap/kvproto/pull/778

### Check List 

Tests 

- Unit test
- Integration test

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM

### Release note 

```release-note
None
```

### CODE REVIEW

File: components/tikv_util/src/codec/bytes.rs [hunk 1]
```diff
@@ -278,6 +278,20 @@ pub fn decode_bytes_in_place(data: &mut Vec<u8>, desc: bool) -> Result<()> {
     }
 }
 
+// Returns the decoded key length of the given encoded key which is assumed valid without actually
+// decoding it.
+pub fn unchecked_decoded_len(encoded: &[u8], desc: bool) -> usize {
```
Comment (breeswish) [inline components/tikv_util/src/codec/bytes.rs:283]: Maybe it's fine to just use the length, since we will at most have a 1/8 * length + 7 inaccuracy.

**Comment by zhongzc** [comment]
Updated, cc @breeswish

**Review by breeswish — APPROVED** [review]

File: src/storage/mvcc/reader/point_getter.rs [hunk 2]
```diff
@@ -554,39 +557,47 @@ mod tests {
...
         let s = getter.take_statistics();
         assert_seek_next_prev(&s.write, 1, 0, 0);
+        assert_eq!(s.processed_size, 0);
 
         // Get a key that exists
         must_get_value(&mut getter, b"foo2", b"foo2v");
         let s = getter.take_statistics();
         // We have to check every version
         assert_seek_next_prev(&s.write, 1, 40, 0);
+        assert_eq!(s.processed_size, 269);
```
Comment (iosmanthus) [inline src/storage/mvcc/reader/point_getter.rs:572]: Could we describe the key structure here? `269` seems too magic and difficult to maintain if the next contributor doesn't familiar with key content.

File: src/storage/mvcc/reader/point_getter.rs [hunk 3]
```diff
@@ -554,39 +557,47 @@ mod tests {
...
         // Get again
         must_get_value(&mut getter, b"foo2", b"foo2v");
         let s = getter.take_statistics();
         assert_seek_next_prev(&s.write, 1, 40, 0);
+        assert_eq!(s.processed_size, 269);
```
Comment (iosmanthus) [inline src/storage/mvcc/reader/point_getter.rs:577]: Ditto. We could use a variable instead. Remember to add some comment to its definition.

**Comment by zhongzc** [comment]
@iosmanthus Updated

**Review by iosmanthus — APPROVED** [review]
LGTM

**Pull request merged**